metagraph/src/common/elias_fano/elias_fano_merger.hpp

```diff
@@ -193,15 +193,15 @@ void merge_files(std::vector<std::string> sources,
         return;
 
     // if there are too many chunks, merge them into larger ones
-    assert(max_sources_open >= 2);
+    assert(max_sources_open >= 3);
     while (sources.size() > max_sources_open) {
         // chunk 0 may be special (e.g. storing only fwd bits) and hence
         // never pre-merged
         std::vector<std::string> new_chunks = { sources.at(0) };
         size_t i = 1;
         while (i < sources.size()) {
             std::vector<std::string> to_merge;
-            while (i < sources.size() && to_merge.size() < max_sources_open) {
+            while (i < sources.size() && to_merge.size() + 1 < max_sources_open) {
```

Comment:
I'm guessing the +1 is because of the output file? Please add a comment to explain this

Reply:
done